corelib/src/util3d_surface.cpp

```diff
@@ -3530,7 +3530,7 @@ LaserScan adjustNormalsToViewPoint(
 
 					float result = v.dot(n);
 					if(result < 0
-					 || (groundNormalsUp>0.0f && ptr[nz] < -groundNormalsUp && ptr[2] < viewpoint[3])) // some far velodyne rays on road can have normals toward ground
```

Comment:
viewpoint is an Eigen::Vector3f object, with indexes 0=x, 1=y and 2=z. 3 would be out of bounds.

Reply:
Thanks for the comment, the other places using viewpoint[3] are wrong. Here the ones I've found:
https://github.com/search?q=repo%3Aintrolab%2Frtabmap%20viewpoint%5B3%5D&type=code

I created an issue about it https://github.com/introlab/rtabmap/issues/1413